src/WebJobs.Script/Workers/Http/IHttpWorkerChannel.cs

```diff
@@ -13,7 +13,7 @@ public interface IHttpWorkerChannel
 
         Task InvokeFunction(ScriptInvocationContext context);
 
-        Task StartWorkerProcessAsync(CancellationToken cancellationToken = default);
+        Task<bool> StartWorkerProcessAsync(CancellationToken cancellationToken = default);
```

Comment:
We should not be relying bool flag here. Ideally IsWorkerReady and pub/sub pattern we have for raising worker error events should take care of this. Can you please take a look at Rpc layer and see if sth is missing in httpworker? 

At somepoint, process management between http worker and Rpc worker should follow common code path. It would be better if they have common interfaces.

Reply:
In RPC, we rely on setting taskCompletionSource _workerInitTask which gets set either during a WorkerInitResponse or during an error scenario (sets exception). The WorkerInitResponse waits on the task which exits with an exception thus resulting resulting in false when checked for .IsCompleted in the continuation task I think. 

I also thought this approach presents a way for the tasks to gracefully exit and not have a shared dependency like the _workerInitTask that is used by both error case and the happy path scenario. We can discuss more about this offline as well.

Reply:
Updated the logic.